File: example/src/main/java/com/github/simonpercic/oklogexample/AppModule.java
package com.github.simonpercic.oklogexample;

import android.content.Context;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * @author Simon Percic <a href="https://github.com/simonpercic">https://github.com/simonpercic</a>
 */
@Module class AppModule {

    private final App app;

    AppModule(App app) {
        this.app = app;
    }

    @Provides @Singleton App provideApp() {
        return app;
    }

    @Provides @Singleton Context provideContext(App app) {
        return app;
    }
}

File: example/src/main/kotlin/com/github/simonpercic/oklogexample/AppModule.kt
package com.github.simonpercic.oklogexample

import android.content.Context
import com.github.simonpercic.oklogexample.logger.FlavorLoggerAppModule
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

/**
 * @author Simon Percic [https://github.com/simonpercic](https://github.com/simonpercic)
 */
@Module(includes = arrayOf(FlavorLoggerAppModule::class))
internal class AppModule(private val app: App) {

    @Provides @Singleton fun provideApp(): App {
        return app
    }

    @Provides @Singleton fun provideContext(app: App): Context {
        return app
    }
}
